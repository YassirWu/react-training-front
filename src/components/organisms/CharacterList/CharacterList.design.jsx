import React from 'react';
import t from 'prop-types';
import { List, ListBody, ListHeader, ListBodyLine } from 'components';

import styles from './CharacterList.module.scss';

const CharacterLine = ({
  id, name, gender, culture,
  books, playedBy,
}) => (
  <ListBodyLine id={id}>
    <td>{name}</td>
    <td>{gender}</td>
    <td>{culture}</td>
    <td>{books}</td>
    <td>{playedBy}</td>
  </ListBodyLine>
);
CharacterLine.propTypes = {
  id: t.number.isRequired,
  name: t.string.isRequired,
  gender: t.string.isRequired,
  culture: t.string.isRequired,
  books: t.number.isRequired,
  playedBy: t.number.isRequired,
};

const renderListLines = character => (
  <CharacterLine
    id={character.id}
    name={character.name}
    gender={character.gender}
    culture={character.culture}
    books={character.books.length}
    playedBy={character.playedBy.length}
  />
);

const CharacterList = ({ characters }) => (
  <List className={styles.characterList}>
    <ListHeader>
      <th>Name</th>
      <th>Gender</th>
      <th>Culture</th>
      <th>Nb of books</th>
      <th>Nb of players</th>
    </ListHeader>
    <ListBody>
      {characters.map(renderListLines)}
    </ListBody>
  </List>
);

CharacterList.defaultProps = {
  characters: [],
};

CharacterList.propTypes = {
  characters: t.arrayOf(
    t.shape({
      name: t.string.isRequired,
      gender: t.string.isRequired,
      culture: t.string.isRequired,
      books: t.arrayOf(t.string.isRequired),
      playedBy: t.arrayOf(t.string.isRequired),
    }),
  ),
};

export default CharacterList;
