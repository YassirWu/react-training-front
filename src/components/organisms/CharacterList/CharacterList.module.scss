.characterList {
  margin-bottom: 15px;
}