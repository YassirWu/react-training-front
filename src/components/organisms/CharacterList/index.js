export { default } from './CharacterList.design';
