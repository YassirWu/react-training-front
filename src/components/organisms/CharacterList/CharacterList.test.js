import React from 'react';
import { shallow } from 'enzyme';

import CharacterList from './CharacterList.design';

describe('<CharacterList />', () => {
  const characters = [
    {
      gander: '978-0553103540',
      culture: 'Hardcover',
      name: 'A Game of Thrones',
      books: [],
      playedBy: []
    },
    {
      gander: '978-0553108033',
      culture: 'Hardback',
      name: 'A Clash of Kings',
      books: [],
      playedBy: []
    },
    {
      gander: '978-0553106633',
      culture: 'Hardcover',
      name: 'A Storm of Swords',
      books: [],
      playedBy: []
    },
    {
      gander: '978-0976401100',
      culture: 'GraphicNovel',
      name: 'The Hedge Knight',
      books: [],
      playedBy: []
    },
  ];
  
  test('should renders CharacterList component', () => {
    const wrapper = shallow(<CharacterList characters={characters} />);
    expect(wrapper).toMatchSnapshot();
  });
  test('should return list of characters', () => {
    const wrapper = shallow(<CharacterList characters={characters} />);
    expect(wrapper.find('CharacterLine')).toHaveLength(4);
  });
});
