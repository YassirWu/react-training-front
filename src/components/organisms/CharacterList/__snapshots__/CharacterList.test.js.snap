// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CharacterList /> should renders CharacterList component 1`] = `ShallowWrapper {}`;
